### PR TITLE
Add binaryen

### PR DESCRIPTION
**Pull request by Warchant**
Signed-off-by: Bogdan Vaneev <warchantua@gmail.com>

* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/Warchant/hunter/builds/22657757
  * https://travis-ci.com/Warchant/hunter/builds/102325636

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/338

---

### CODE REVIEW

File: docs/packages/pkg/binaryen.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+.. _pkg.binaryen:
+
+Binaryen
+========
+
+-  `Official <https://github.com/WebAssembly/binaryen/>`__
+-  `Hunterized <https://github.com/warchant/binaryen/>`__
+-  `Example <https://github.com/ruslo/hunter/blob/master/examples/binaryen/CMakeLists.txt>`__
+-  Added by `Warchant <https://github.com/Warchant>`__ (`pr-N <https://github.com/ruslo/hunter/pull/N>`__)
```
Comment (ruslo) [inline docs/packages/pkg/binaryen.rst]: `s,N,1751,`

**Comment by Warchant** [comment]
Yes, PR is WIP, because upstream CI failing, I want to fix it first.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/22660308
- https://travis-ci.org/ingenue/hunter/builds/498760107?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
- https://ci.appveyor.com/project/ingenue/hunter/builds/22661540
- https://travis-ci.org/ingenue/hunter/builds/498779560?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.118

**Pull request merged**